### DISCUSSION TITLE
What is the reason to cap Athena GetQueryResults.MaxResults as 1000

### DISCUSSION THREAD

**Discussion by reenimathew** [aws/aws-sdk-java#3094] [Q&A]
We want to use the Pagination feature present in AWS SDK2 for Athena. But 1000 is way too less for us.
Is it possible to override this value?
I see that the documentation caps 1000

 `[MaxResults](https://docs.aws.amazon.com/athena/latest/APIReference/API_GetQueryResults.html#API_GetQueryResults_RequestSyntax)
The maximum number of results (rows) to return in this request.

Type: Integer
Valid Range: Minimum value of 1. Maximum value of 1000.
Required: No`

**Comment by debora-ito** [comment]
That is a max value set by the service team, there's no way to increase it from the client side.

**Reply by debora-ito** [reply]
The Java SDK v2 provides automatic pagination though, you can use it for a seamless pagination experience - https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/pagination.html
